bp_common/test/src/perch/bp_utils.h

```diff
@@ -18,4 +18,43 @@ void bp_finish(uint8_t code);
 
 #define BP_CFG_BASE_ADDR ((char *)(0x00200000))
 
+/////////////////////////accelerator_vector_dot_product////////////////////
+#define CACCEL_VDP_BASE_ADDR  ((uint64_t *)(0x04000000))
+#define CACCEL_VADD_BASE_ADDR ((uint64_t *)(0x05000000))
+#define SACCEL_VDP_BASE_ADDR  ((uint64_t *)(0x06000000))
+#define SACCEL_VADD_BASE_ADDR ((uint64_t *)(0x07000000))
+#define SACCEL_VDP_MEM_BASE   ((uint64_t *)(0x1000000000))
+#define SACCEL_VADD_MEM_BASE  ((uint64_t *)(0x1800000000))
+
+#define ACCEL_VPD_INPUT_A_PTR    0
+#define ACCEL_VPD_INPUT_B_PTR    1
+#define ACCEL_VPD_INPUT_LEN      2
+#define ACCEL_VPD_START_CMD      3
+#define ACCEL_VPD_RESP_STATUS    4
+#define ACCEL_VPD_RESP_PTR       5
+#define ACCEL_VPD_RESP_LEN       6
+#define ACCEL_VPD_OPERATION      7
+#define ACCEL_VPD_NUM_CSRs       8
+
+struct VDP_CSR
+{ 
+  uint64_t *input_a_ptr;
+  uint64_t *input_b_ptr;
+  uint64_t input_length;
+  uint64_t *resp_ptr;
+};
+
+void bp_set_CSR(uint64_t *accel_base_address, uint8_t csr_idx, uint64_t csr_value);
+uint64_t bp_get_CSR(uint64_t *accel_base_address, uint8_t csr_idx); 
+void mem_cpy(uint64_t *src, uint64_t *dest, uint64_t length);
```

Comment:
Can't we use the normal memcpy from bsg_newlib? If that's too heavyweight, maybe rename to something closer to dma